### PR TITLE
[Phi] Move grid sample op kernel into phi

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

Function optimization
### PR changes

OPs
### Describe

[Phi] Move grid sample op kernel into phi

### CODE REVIEW

**Review by MingMingShangTian — APPROVED** [review]
LGTM

File: paddle/phi/kernels/cpu/grid_sample_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  }
+}
+
+template <typename T>
+static void CalcGridLocations(const CPUContext& ctx,
+                              const DenseTensor& grid,
+                              const int in_h,
+                              const int in_w,
+                              bool align_corners,
+                              std::string padding_mode,
```
Comment (zyfncg) [inline paddle/phi/kernels/cpu/grid_sample_kernel.cc:64]: 这里用`const string&` 会不会更高效一些？

**Review by zyfncg — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

File: paddle/phi/kernels/cpu/grid_sample_kernel.cc (hunk 1 above)
Comment (chenwhql) [reply paddle/phi/kernels/cpu/grid_sample_kernel.cc:64]: 嗯嗯，这里原先代码未做改动，后续完善

**Review by YuanRisheng — APPROVED** [review]

**Pull request merged**